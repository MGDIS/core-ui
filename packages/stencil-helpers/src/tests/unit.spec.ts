import { beforeEach, describe, expect, test, vi } from 'vitest';
import { setupMutationObserverMock, setUpRequestAnimationFrameMock, setupResizeObserverMock, setupSubmitEventMock } from '.';

describe('test - e2e', () => {
  describe('setupMutationObserverMock', () => {
    test.each([{}, { disconnect: vi.fn(), observe: ({}) => {}, takeRecords: () => [] }])('Should render html', args => {
      const Mock = setupMutationObserverMock(args);
      expect(new Mock(args)).toMatchSnapshot();
    });
  });
  describe('setupResizeObserverMock', () => {
    test.each([{}, { disconnect: vi.fn(), observe: ({}) => {}, unobserve: () => [] }])('Should render html', args => {
      const Mock = setupResizeObserverMock(args);
      expect(new Mock(args)).toMatchSnapshot();
    });
  });
  describe('setupSubmitEventMock', () => {
    test('Should render html', () => {
      const Mock = setupSubmitEventMock();
      expect(new Mock({})).toMatchSnapshot();
    });
  });
  describe('setUpRequestAnimationFrameMock', () => {
    beforeEach(() => {
      vi.useFakeTimers();
    });

    test.each([window, global])('Should render html', element => {
      const timer = vi.fn();
      const callback = vi.fn();

      setUpRequestAnimationFrameMock(timer);

      expect(element.requestAnimationFrame).toBeDefined();

      element.requestAnimationFrame(callback);
      vi.runOnlyPendingTimers();

      expect(timer).toHaveBeenCalled();
      expect(callback).toHaveBeenCalled();
    });
  });
});
