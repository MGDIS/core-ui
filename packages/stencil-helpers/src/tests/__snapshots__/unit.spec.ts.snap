// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test - e2e > setUpRequestAnimationFrameMock > Should render html 1`] = `0`;

exports[`test - e2e > setupMutationObserverMock > Should render html 1`] = `
MockMutationObserver {
  "cb": {},
  "disconnect": undefined,
  "observe": undefined,
  "takeRecords": undefined,
}
`;

exports[`test - e2e > setupMutationObserverMock > Should render html 2`] = `
MockMutationObserver {
  "cb": {
    "disconnect": [MockFunction spy],
    "observe": [Function],
    "takeRecords": [Function],
  },
  "disconnect": [MockFunction spy],
  "observe": [Function],
  "takeRecords": [Function],
}
`;

exports[`test - e2e > setupResizeObserverMock > Should render html 1`] = `
MockResizeObserver {
  "cb": {},
  "disconnect": undefined,
  "observe": undefined,
  "unobserve": undefined,
}
`;

exports[`test - e2e > setupResizeObserverMock > Should render html 2`] = `
MockResizeObserver {
  "cb": {
    "disconnect": [MockFunction spy],
    "observe": [Function],
    "unobserve": [Function],
  },
  "disconnect": [MockFunction spy],
  "observe": [Function],
  "unobserve": undefined,
}
`;

exports[`test - e2e > setupSubmitEventMock > Should render html 1`] = `
SubmitEvent {
  "detail": undefined,
  "isTrusted": false,
}
`;
